### PR TITLE
Allow setting uploader timeout to `nil`

### LINKED ISSUE #384: Disable timeout for RestClient
While using the gem, we can pass `timeout` in `options` which is further used while making request using `RestClient`.

Example: https://github.com/cloudinary/cloudinary_gem/blob/d84227247d6ff1f9cee1a069ddc892e548854798/lib/cloudinary/uploader.rb#L320

To disable timeouts, a value of `-1` or `nil` must be provided as timeout to RestClient.
`-1` has been deprecated by RestClient and instead it is advised to use `nil` to disable timeout.
https://github.com/rest-client/rest-client/blob/2c72a2e77e2e87d25ff38feba0cf048d51bd5eca/lib/restclient/request.rb#L708

If we use `nil` from client side, Cloudinary gem will use the default of 60 seconds.
Please add support for disabling timeouts for API calls.
The processing time of rename API call is dependent upon number of transformations and usually exceeds the default timeout of 60 seconds.

The `nil` timeout should be supported via both `options` param in API calls as well as through `Cloudinary.config.timeout`

### PR DESCRIPTION
**Pull request by const-cloudinary**
Fixes #384

### CODE REVIEW

File: lib/cloudinary/uploader.rb [hunk 1]
```diff
@@ -317,7 +317,7 @@ def self.call_api(action, options)
       params[:signature] = Cloudinary::Utils.api_sign_request(params.reject { |k, v| non_signable.include?(k) }, api_secret)
       params[:api_key]   = api_key
     end
-    timeout = options[:timeout] || Cloudinary.config.timeout || 60
+    timeout = options.fetch(:timeout, Cloudinary.config.to_h.fetch(:timeout, 60))
```
Comment (amcaplan) [inline lib/cloudinary/uploader.rb:320]:
```suggestion
    timeout = options.fetch(:timeout) { Cloudinary.config.to_h.fetch(:timeout, 60) }
```

The block is lazily evaluated, unlike when it's simply an argument in parentheses. This saves us the need to create a new hash copy of `Cloudinary.config` when it's not necessary.

File: spec/uploader_spec.rb [hunk 2]
```diff
@@ -385,6 +385,21 @@
...
+
+    it "should allow passing nil value" do
+      expect(RestClient::Request).to receive(:execute) do |args|
+        expect(args[:timeout]).to be nil
+      end
+      Cloudinary::Uploader.upload(TEST_IMG, :timeout => nil)
+    end
+
+    it "should fall back to default timeout" do
+      Cloudinary.config.delete_field(:timeout)
```
Comment (amcaplan) [inline spec/uploader_spec.rb:397]: Maybe want to reset the state of `Cloudinary.config` after the test?

File: spec/uploader_spec.rb [hunk 3]
```diff
@@ -385,6 +385,21 @@
     it "should fail if timeout is reached" do
       expect{Cloudinary::Uploader.upload(Pathname.new(TEST_IMG), :tags => [TEST_TAG, TIMESTAMP_TAG])}.to raise_error(RestClient::RequestTimeout)
     end
+
+    it "should allow passing nil value" do
```
Comment (amcaplan) [inline spec/uploader_spec.rb:389]: This assumes a certain initial state of `Cloudinary.config`, maybe want to make that explicit?

File: spec/uploader_spec.rb (hunk 2 above)
Comment (const-cloudinary) [reply spec/uploader_spec.rb:397]: this test is under   `describe ":timeout" do` on line 376

File: spec/uploader_spec.rb (hunk 3 above)
Comment (const-cloudinary) [reply spec/uploader_spec.rb:389]: We first check immediate options and if its not there, fall back to config, and config is set to something not `nil` on line 379

**Pull request merged**